addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -477,6 +477,46 @@ def test_larger_invoice_auto_reconcile(self):
 
         self.assertEqual(self.invoice_line_1.amount_residual, 60.0, "The invoice should have been partially reconciled")
 
+    def test_auto_reconcile_with_duplicate_match(self):
+        """ If multiple bank statement lines match with the same invoice, ensure the
+         correct line is auto-validated and no crashing happens.
+        """
+
+        # Only the invoice defined in this test should have this partner
+        partner = self.env['res.partner'].create({'name': "The Only One"})
+        invoice_line = self._create_invoice_line(
+            2000, partner, 'out_invoice', ref="REF 7788")
+
+        # Enable auto-validation and don't restrict the partners that can be matched on
+        # so our newly created partner can be matched.
+        self.rule_1.write(
+            {'auto_reconcile': True,
+             'match_partner_ids': [(5, 0, 0)],
+             }
+        )
```

Comment:
Not a big fan of this indentation at all

```suggestion
        self.rule_1.write({
            'auto_reconcile': True,
            'match_partner_ids': [(5, 0, 0)],
        })
```

Because if for some reason the name `rule_1` changes (more precisely its length), you will have to adapt all the following lines instead of just one.

I like to either inline, either put one thing per line. In this case, I am inlining the arguments of `write` but putting one key/value per line of the dict. When I spread a call on multiple lines, the opening and closing are on separate lines always.

Reply:
Uh, my explanation works for `_check_statement_matching` below :roll_eyes:

Reply:
I agree, I thought about reformatting it but forgot.

Reply:
The dangers of relying on code formatting tools and then not using them ;-). I've updated the PR.